src/net/parse.cpp

```diff
@@ -47,6 +47,12 @@ namespace net
                 port = address.substr(address.rfind(':') + 1);
             }
         }
+        // Else if IPv6 address format without port e.g. "addr:addr:addr:...:addr"
+        else if (std::count(address.begin(), address.end(), ':') > 1)
```

Comment:
This is the main one worth doing, use the builtins!

Reply:
Wait this is a `>1`, let me check the spec on that.

Reply:
That doesn't do the same though: this line is asking for *more than* 1 match since all valid IPv6 address must have at least 2 colons.

Reply:
> That doesn't do the same though: this line is asking for more than 1 match since all valid IPv6 address must have at least 2 colons.

I was replying to myself during this thread because I couldn't see your responses in real time. So, I simultaneously writing the same thing. Weirdly quick on your part though to respond! I need to set better email filtering/alerts myself.

Reply:
Haha no worries! You just so happened to catch me during a rare time where I was reading my emails lol